db/migrations/versions/009_add_binary_scoring_system.py

```diff
@@ -0,0 +1,31 @@
+"""Adds a binary scoring system to allow a simple accept/reject option
+
+Revision ID: "009_add_binary_scoring_system"
+Revises: "008_extend_flag_for_change_req"
+Create Date: 2024-12-19 16:30:00.346479
+
+"""
+
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "009_add_binary_scoring_system"
+down_revision = "008_extend_flag_for_change_req"
+branch_labels = None
+depends_on = None
+
+zero_to_one_id = "40d10744-b0c1-4af0-9cb9-df1058a3e930"
+
+
+def upgrade():
+    op.execute(
+        sa.text(
+            "INSERT INTO scoring_system (id, scoring_system_name, minimum_score, maximum_score) "
+            "VALUES (:id, 'ZeroToOne', 0, 1)"
+        ).bindparams(id=zero_to_one_id)
+    )
```

Comment:
Please can we not use alembic migrations for inserting data - this can cause problems over time and it often leads to code that is tightly intermingled with the expectation of certain data. I think we'd like to avoid this going forwards.

[Alembic migrations should be used for structural changes only, ideally](https://github.com/sqlalchemy/alembic/discussions/972#discussioncomment-1766394). Occasionally I think it _can_ be OK to do UPDATEs that affect existing data to wrangle them into the right shape, but nothing that adds new data from scratch.

A good rule of thumb to keep is that after all the alembic migrations have run, we should always have an empty database.

If we need to insert a new entry here, can we add a script or invoke task that gets manually run to insert the data for now please? 🙏

Reply:
Also - this scoring system should be added to the local seeding script here: https://github.com/communitiesuk/funding-service-pre-award-stores/blob/27e15e24de3d0e9fa5a2089ac19215a1d9c5c6fc/assessment_store/tasks/db_tasks.py#L127

That's how we get data into the local DB.

It doesn't solve the "how do we get data into the live DBs" - but we need to do that separately. I would suggest that until we build a robust answer, maybe just write a script that can be run against an environment to insert it?

Reply:
Thanks for having a look at this! I was a bit unsure how we're approaching things in the "new monolithic world" so I just followed the existing patterns that we've been using.  

I do agree in principal that migrations aren't suited to adding data, but as in the discussion you linked, I'm not sure there is a definitively better alternative. Personally I think adding a scoring system is fairly safe since it's essentially seed data and really it shouldn't even have a downgrade (even though I've provided it, perhaps I shouldn't have). 

But I'm happy to create a script instead until we've had a chance to agree a long-term solution. Do we currently auto run any data scripts as part of deployment or is it something I'll need to do manually?

Reply:
We don't have anything that would auto run the data migrations, yeah - so it would just be a manual task. If you add a script somewhere then it should be a case of something like `copilot svc exec --name fsd-pre-award --command "launcher python <scriptname>` - so not super overly arduous (but you have to do it for each environment, and consider how you're deploying things so that the first deploy gets the script in the environment but can survive without the data, etc).

Reply:
@hamzabinkhalid is adding the scoring system as part of his PR, so I've removed it from mine and referred him to this discussion.